lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -105,6 +106,9 @@ func InitializeImageCache(client *client.PortLayer) error {
 
 		// populate the trie with IDs
 		for k := range i.CacheByID {
+			// Trim the "sha256:" prefix from the CacheByID key before indexing iDIndex
+			// as it is keyed by the full image ID without the "sha256:" prefix.
+			k = strings.TrimPrefix(k, sha256Prefix)
```

Comment:
We are currently exclusively using `sha256` but the only reason for there to be a prefix is to allow for other hashes to be used.
I'd suggest `strings.SplitN(k, ":", 2)` or similar and use the last element.

Reply:
Fixed!